pkg/authentication/gcp_access_token.go

```diff
@@ -0,0 +1,80 @@
+package authentication
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"time"
+
+	"google.golang.org/api/iam/v1"
+	"google.golang.org/api/option"
+)
+
+// GCPIamServiceAccountAccessToken is a higher order function that returns an
+// AccessTokenAuthentication func.
+// The resulting function allows verifying access tokens provided by GCP IAM for
+// service accounts.
+func GCPIamServiceAccountAccessToken(project, serviceAccountName string) AccessTokenAuthentication {
+	svc, err := newIamService()
```

Comment:
Since this doesn't pass in a context, consider making this a package var to avoid having to instantiate a new service for each request.

Reply:
We're only initializing the service once, as we're returning a function that will be used by the interceptor.